### PR TITLE
Added Array benchmarks

### PR DESCRIPTION
**Pull request by Tolstya4ok**

### CODE REVIEW

File: benchmarks/implementations/joint/bindings/python/Component.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import pyjoint
 from .Benchmarks_adapters import *
+from builtins import range
```
Comment (koplyarov) [inline benchmarks/implementations/joint/bindings/python/Component.py]: Add future:builtins to REQUIRED_PYMODULES in scripts/check_dependencies.sh

File: benchmarks/templates/array.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# C
+|               | array.Get(), ns | array.Set(), ns |
+| ------------- | ------------: | --------: |
+| Native call | - | - |
+| Proxy-side call | ${array.proxy_array_get.joint(lang:c)[main]} | ${array.proxy_array_set.joint(lang:c)[main]} |
+
+# C++
```
Comment (koplyarov) [inline benchmarks/templates/array.md]: Add noexcept C++ version (see enum.md)

Comment (koplyarov) [inline benchmarks/templates/array.md]: Put C++ benchmarks before C. This is not cool, but we have to keep the order consistent between different templates.

File: benchmarks/benchmarks/Array.hpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    public:
+        ArrayBenchmarks()
+            : Base("array", [](const BenchmarkCtx& ctx) { return ctx.CreateArrayBenchmarks(); })
+        {
+            using i64 = int64_t;
+            using C = typename Desc_::BenchmarkCtx;
+            using B = typename Desc_::ArrayBenchmarksPtr;
+
+            AddSimpleBenchmark("native_array_get", [](C& ctx, i64 n, B b){ CALL( b->MeasureNativeGetI64Element(n) ); });
```
[the remark this answers is not in the document]
Comment (Tolstya4ok) [reply benchmarks/benchmarks/Array.hpp]: done

File: benchmarks/implementations/joint/bindings/c/Benchmarks.c [hunk 4]
```diff
@@ -36,6 +37,9 @@ JointCore_Error Benchmarks_Init(Benchmarks* self, Joint_ModuleContext moduleCont
 
     self->moduleContext = moduleContext;
 
+    JointCore_Exception_Handle ex;
+    i64__Array_Create(100, &(self->array), &ex);
```
[the remark this answers is not in the document]
Comment (Tolstya4ok) [reply benchmarks/implementations/joint/bindings/c/Benchmarks.c]: done

File: benchmarks/implementations/joint/bindings/c/Benchmarks.c [hunk 5]
```diff
@@ -49,6 +53,44 @@ JointCore_Error Benchmarks_Deinit(Benchmarks* self)
 }
 
 
+///// IArrayBenchmarks /////
+
+JointCore_Error Benchmarks_MeasureGetI64Element(Benchmarks* self, int64_t n, JointCore_Exception_Handle* ex)
+{
+    int64_t i;
+    int64_t result = 0;
+    int64_t retval = 0;
```
[the remark this answers is not in the document]
Comment (Tolstya4ok) [reply benchmarks/implementations/joint/bindings/c/Benchmarks.c]: done

File: benchmarks/implementations/joint/bindings/c/Benchmarks.c [hunk 6]
```diff
@@ -49,6 +53,44 @@ JointCore_Error Benchmarks_Deinit(Benchmarks* self)
...
+            return retval;
+    }
+
+    return JOINT_CORE_ERROR_NONE;
+}
+
+JointCore_Error Benchmarks_MeasureSetI64Element(Benchmarks* self, int64_t n, JointCore_Exception_Handle* ex)
+{
+    int64_t i;
+    int64_t retval = 0;
```
[the remark this answers is not in the document]
Comment (Tolstya4ok) [reply benchmarks/implementations/joint/bindings/c/Benchmarks.c]: done

File: benchmarks/templates/array.md (hunk 2 above)
Comment (Tolstya4ok) [reply benchmarks/templates/array.md]: Okay

Comment (Tolstya4ok) [reply benchmarks/templates/array.md]: done

File: benchmarks/implementations/joint/bindings/python/Component.py (hunk 1 above)
Comment (Tolstya4ok) [reply benchmarks/implementations/joint/bindings/python/Component.py]: done

File: benchmarks/implementations/joint/bindings/c/Benchmarks.c [hunk 7]
```diff
@@ -38,7 +38,10 @@ JointCore_Error Benchmarks_Init(Benchmarks* self, Joint_ModuleContext moduleCont
     self->moduleContext = moduleContext;
 
     JointCore_Exception_Handle ex;
-    i64__Array_Create(100, &(self->array), &ex);
+
+    int64_t retval = i64__Array_Create(100, &(self->array), &ex);
```
Comment (koplyarov) [inline benchmarks/implementations/joint/bindings/c/Benchmarks.c]: JointCore_Error!!1

File: benchmarks/implementations/joint/bindings/c/Benchmarks.c [hunk 8]
```diff
@@ -62,7 +62,7 @@ JointCore_Error Benchmarks_MeasureGetI64Element(Benchmarks* self, int64_t n, Joi
 {
     int64_t i;
     int64_t result = 0;
-    int64_t retval = 0;
+    int64_t retval = JOINT_CORE_ERROR_NONE;
```
Comment (koplyarov) [inline benchmarks/implementations/joint/bindings/c/Benchmarks.c]: JointCore_Error

Comment (Tolstya4ok) [reply benchmarks/implementations/joint/bindings/c/Benchmarks.c]: Makes sense

File: benchmarks/implementations/joint/bindings/c/Benchmarks.c (hunk 7 above)
Comment (Tolstya4ok) [reply benchmarks/implementations/joint/bindings/c/Benchmarks.c]: Fixed!!1

**Pull request merged**